rfc/java17.md

```diff
@@ -0,0 +1,183 @@
+| **Author(s)**  | Matt Gogerly ([`@mattgogerly`](https://github.com/mattgogerly)) |
+| **SIG / WG**   | Platform SIG |
+
+## Overview
+
+To keep current with the evolution of Java, Spinnaker needs to move from Java 11
+to Java 17. This RFC documents a process and timeline for doing so.
+
+### Goals and Non-Goals
+
+Goals:
+
+*   Provide a process for moving official Spinnaker builds to the Java 17 JRE
+*   Discuss ways we will encourage people using non-offical builds to also move
+    to the Java 17 JRE
+*   Provide a timeline for the final switch to a Java 17 JDK along with criteria
+    to decide if the timeline should be delayed
+
+Non-Goals:
+
+*   Providing a rationale for upgrading Java; all active Java projects need to
+    update their JDK version eventually as libraries we depend upon will also
+    depend on new versions of the JDK
+*   Debating which Java version we should upgrade to; we take it as a given that
+    only LTS releases are acceptable for most organizations, especially for
+    running critical infrastructure like Spinnaker
+*   Documenting and solving technical issues that present themselves in the
+    migration; those will be handled separately, but this document concerns
+    itself primarily with the larger processes and timeline
+*   Detailing the required release system changes; since the releases are still
+    entirely maintained by Google, these changes aren't particularly relevant to
+    the wider community
+
+### JRE vs. JDK
+
+There are two components of Java of interest to this RFC. The Java Runtime
+Environment (JRE) is the `java` binary and associated libraries used to _run_
+Spinnaker services. The Java Development Kit (JDK) is the `javac` binary and
+associated libraries used to _compile_ Spinnaker.
+
+Once we upgrade to the Java 17 JDK, Spinnaker will no longer be able to run in a
+Java 11 JRE. So the first (more complicated) step is to get everyone migrated to
+a Java 17 JRE. Only after that is done can we switch to the Java 17 JDK.
+
+For official Spinnaker containers, we control the JRE used to run Spinnaker. For
+those releases, this proposal takes a relatively slow ramp to the Java 17 JRE
+and provides an escape hatch back to Java 8 for users who experience problems.
+
+There are also some unknown number of customers who aren't using the Spinnaker-
+provided containers. We will need to do some outreach to make sure these users
+convert to a Java 17 JRE too.
+
+## Timeline
+
+The timeline in which this was implemented is detailed below.
+
+### 1.31 (early March)
+
+Before we can upgrade to Java 17 we must first upgrade from Gradle 6 to Gradle 7.
+The [compatibility matrix](https://docs.gradle.org/current/userguide/compatibility.html)
+shows that Java 17 is supported by Gradle 7.3, however in the interests of not
+adding more technical debt this RFC proposes we upgrade to the latest Gradle, 7.6.
+
+Early testing suggests this shouldn't be too much of a hassle - I was able to compile
```

Comment:
That's great.  I suspect the most difficult project to migrate to gradle 7 is https://github.com/spinnaker/spinnaker-gradle-project.